packages/mermaid/src/diagrams/flowchart/flowDb.ts

```diff
@@ -16,1101 +18,1082 @@ import {
-    if (txt.startsWith('"') && txt.endsWith('"')) {
-      txt = txt.substring(1, txt.length - 1);
+  /**
+   * Function called by parser when a node definition has been found
+   */
+  public addVertex = (
+    id: string,
+    textObj: FlowText,
+    type: FlowVertexTypeParam,
+    style: string[],
+    classes: string[],
+    dir: string,
+    props = {},
+    metadata: any
+  ) => {
+    if (!id || id.trim().length === 0) {
+      return;
     }
-    vertex.text = txt;
-  } else {
-    if (vertex.text === undefined) {
-      vertex.text = id;
+    // Extract the metadata from the shapeData, the syntax for adding metadata for nodes and edges is the same
+    // so at this point we don't know if it's a node or an edge, but we can still extract the metadata
+    let doc;
+    if (metadata !== undefined) {
+      let yamlData;
+      // detect if shapeData contains a newline character
+      if (!metadata.includes('\n')) {
+        yamlData = '{\n' + metadata + '\n}';
+      } else {
+        yamlData = metadata + '\n';
+      }
+      doc = yaml.load(yamlData, { schema: yaml.JSON_SCHEMA }) as NodeMetaData;
     }
-  }
-  if (type !== undefined) {
-    vertex.type = type;
-  }
-  if (style !== undefined && style !== null) {
-    style.forEach(function (s) {
-      vertex.styles.push(s);
-    });
-  }
-  if (classes !== undefined && classes !== null) {
-    classes.forEach(function (s) {
-      vertex.classes.push(s);
-    });
-  }
-  if (dir !== undefined) {
-    vertex.dir = dir;
-  }
-  if (vertex.props === undefined) {
-    vertex.props = props;
-  } else if (props !== undefined) {
-    Object.assign(vertex.props, props);
-  }
 
-  if (doc !== undefined) {
-    if (doc.shape) {
-      if (doc.shape !== doc.shape.toLowerCase() || doc.shape.includes('_')) {
-        throw new Error(`No such shape: ${doc.shape}. Shape names should be lowercase.`);
-      } else if (!isValidShape(doc.shape)) {
-        throw new Error(`No such shape: ${doc.shape}.`);
+    // Check if this is an edge
+    const edge = this.edges.find((e) => e.id === id);
+    if (edge) {
+      const edgeDoc = doc as EdgeMetaData;
+      if (edgeDoc?.animate) {
+        edge.animate = edgeDoc.animate;
       }
-      vertex.type = doc?.shape;
+      if (edgeDoc?.animation) {
+        edge.animation = edgeDoc.animation;
+      }
+      return;
     }
 
-    if (doc?.label) {
-      vertex.text = doc?.label;
+    let txt;
+
+    let vertex = this.vertices.get(id);
+    if (vertex === undefined) {
+      vertex = {
+        id,
+        labelType: 'text',
+        domId: MERMAID_DOM_ID_PREFIX + id + '-' + this.vertexCounter,
+        styles: [],
+        classes: [],
+      };
+      this.vertices.set(id, vertex);
     }
-    if (doc?.icon) {
-      vertex.icon = doc?.icon;
-      if (!doc.label?.trim() && vertex.text === id) {
-        vertex.text = '';
+    this.vertexCounter++;
+
+    if (textObj !== undefined) {
+      this.config = getConfig();
```

Comment:
Why do we need to update the config when adding every vertex?

I see this is from the old code, so you can leave it in now, and fix later.